### ISSUE TITLE
Finish talk-recording geckodriver install or use chromium

### ISSUE THREAD

**Issue by szaimen** [nextcloud/all-in-one#2676]
Things that should be done after https://github.com/nextcloud/all-in-one/pull/2645 is finished
- needs https://github.com/nextcloud/all-in-one/issues/3570
- [x] ~~test if using the internal signaling IP also works~~ does not work
- [x] ~~try if chromium works after https://github.com/nextcloud/spreed/pull/9174. If not, see below (should be available with 28)~~ not recommended 

geckodriver:
- [x] wait for https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/56841 being merged
  - [x] then change the `apk add repository edge/testing` to `apk add repository edge/community` https://github.com/nextcloud/all-in-one/pull/3907
- [ ] wait for alpine v3.xx.1 docker image release (maybe v3.20.1?)
  - [ ] update talk image to v3.xx.1
  - [ ] then move geckodriver to normal `apk add` line and remove `apk add repository edge/community` line

**Comment by Zoey2936** [comment]
can I improve it a bit?

**Comment by szaimen** [comment]
Possibly we could alternatively as soon as chromium support comes with https://github.com/nextcloud/spreed/pulls?q=is%3Apr+is%3Aopen+chromium switch to that...

**Comment by Zoey2936** [comment]
are there differences between chromium and firefox?

**Comment by szaimen** [comment]
So apparently the recording server is reachable but it does not record. I get these logs now:
```
console.warn: services.settings: Ignoring preference override of remote settings server
console.warn: services.settings: Allow by setting MOZ_REMOTE_SETTINGS_DEVTOOLS=1 in the environment
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libEGL initialize failed (t=1.40348) [GFX1-]: glxtest: libEGL initialize failed
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libEGL initialize failed (t=1.40348) |[1][GFX1-]: glxtest: GLX extension missing (t=1.40355) [GFX1-]: glxtest: GLX extension missing
1686043005688   Marionette      INFO    Marionette enabled
1686043005695   Marionette      INFO    Listening on port 35609
Read port: 35609
console.error: "Warning: unrecognized command line flag -width\n"
console.error: "Warning: unrecognized command line flag -height\n"
WebDriver BiDi listening on ws://127.0.0.1:41207
1686043006011   RemoteAgent     WARN    TLS certificate errors will be ignored for this session
console.error: ({})
DevTools listening on ws://127.0.0.1:41207/devtools/browser/a8bb929e-2b73-4c8e-99ab-2356c14d4c12
1686043008674   RemoteAgent     INFO    Perform WebSocket upgrade for incoming connection from 127.0.0.1:50114
1686043030725   Marionette      INFO    Stopped listening on port 35609
console.error:
  NotFoundError: No such JSProcessActor 'WebDriverProcessData': get processActor@chrome://remote/content/webdriver-bidi/modules/WindowGlobalBiDiModule.sys.mjs:16:40
get nodeCache@chrome://remote/content/webdriver-bidi/modules/WindowGlobalBiDiModule.sys.mjs:12:5
#onConsoleAPIMessage@chrome://remote/content/webdriver-bidi/modules/windowglobal/log.sys.mjs:141:23
emit@resource://gre/modules/EventEmitter.sys.mjs:154:20
#onConsoleAPIMessage@chrome://remote/content/shared/listeners/ConsoleAPIListener.sys.mjs:119:10
CS_recordEvent@resource://gre/modules/ConsoleAPIStorage.jsm:172:19
beforeMount/e/</<@https://aio-testing.nextcloud.com/custom_apps/spreed/js/talk-recording.js?v=959ec500-0:2:6003429

JavaScript error: resource://gre/actors/AudioPlaybackChild.sys.mjs, line 17: InvalidStateError: JSWindowActorChild.sendAsyncMessage: JSWindowActorChild cannot send at the moment
1686043167629   geckodriver     INFO    Listening on 127.0.0.1:52471
1686043168199   mozrunner::runner       INFO    Running command: MOZ_CRASHREPORTER="1" MOZ_CRASHREPORTER_NO_REPORT="1" MOZ_CRASHREPORTER_SHUTDOWN="1" MOZ_NO_REMOTE="1" "/usr ... 80" "--remote-debugging-port" "54207" "--remote-allow-hosts" "localhost" "-no-remote" "-profile" "/tmp/rust_mozprofile3Dj5sQ"
console.warn: services.settings: Ignoring preference override of remote settings server
console.warn: services.settings: Allow by setting MOZ_REMOTE_SETTINGS_DEVTOOLS=1 in the environment
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libEGL initialize failed (t=0.74969) [GFX1-]: glxtest: libEGL initialize failed
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: glxtest: libEGL initialize failed (t=0.74969) |[1][GFX1-]: glxtest: GLX extension missing (t=0.749767) [GFX1-]: glxtest: GLX extension missing
1686043168976   Marionette      INFO    Marionette enabled
1686043168982   Marionette      INFO    Listening on port 36985
Read port: 36985
console.error: "Warning: unrecognized command line flag -width\n"
console.error: "Warning: unrecognized command line flag -height\n"
WebDriver BiDi listening on ws://127.0.0.1:54207
1686043169281   RemoteAgent     WARN    TLS certificate errors will be ignored for this session
console.error: ({})
DevTools listening on ws://127.0.0.1:54207/devtools/browser/7a770e78-012d-4aa4-a3ec-4a1359d706a8
1686043171888   RemoteAgent     INFO    Perform WebSocket upgrade for incoming connection from 127.0.0.1:54722
console.error: "update.locale" " file doesn't exist in either the application or GRE directories"
1686043214274   Marionette      INFO    Stopped listening on port 36985
console.error:
  NotFoundError: No such JSProcessActor 'WebDriverProcessData': get processActor@chrome://remote/content/webdriver-bidi/modules/WindowGlobalBiDiModule.sys.mjs:16:40
get nodeCache@chrome://remote/content/webdriver-bidi/modules/WindowGlobalBiDiModule.sys.mjs:12:5
#onConsoleAPIMessage@chrome://remote/content/webdriver-bidi/modules/windowglobal/log.sys.mjs:141:23
emit@resource://gre/modules/EventEmitter.sys.mjs:154:20
#onConsoleAPIMessage@chrome://remote/content/shared/listeners/ConsoleAPIListener.sys.mjs:119:10
CS_recordEvent@resource://gre/modules/ConsoleAPIStorage.jsm:172:19
beforeMount/e/</<@https://aio-testing.nextcloud.com/custom_apps/spreed/js/talk-recording.js?v=959ec500-0:2:6003429

JavaScript error: resource://gre/actors/AudioPlaybackChild.sys.mjs, line 17: InvalidStateError: JSWindowActorChild.sendAsyncMessage: JSWindowActorChild cannot send at the moment
```

**Comment by szaimen** [comment]
Apparently the logs are unrelated. The recording worked nonetheless. But it was not automatically shared into the conversation, I only got a notification.

**Comment by szaimen** [comment]
> * test if using the internal signaling IP also works

Does not work

**Comment by Zoey2936** [comment]
> Apparently the logs are unrelated. The recording worked nonetheless. But it was not automatically shared into the conversation, I only got a notification.

so it works?

**Comment by szaimen** [comment]
> so it works?

yes :)

**Comment by Zoey2936** [comment]
do you now create the issue on the aports??

**Comment by Zoey2936** [comment]
The issue was closed...
So what do you to do now?
do you want to switch to chrome then or should I ask if I can take over the maintance of geckodriver?

**Comment by szaimen** [comment]
> The issue was closed...
> So what do you to do now?
> do you want to switch to chrome then or should I ask if I can take over the maintance of geckodriver?

I would say lets wait for https://github.com/nextcloud/spreed/pull/9174 to be merged and released. Then update the container to use chromium. If this works, I would say maintaining upstream geckodriver is not needed. If we cannot make it work with chromium then we can think about further steps. And thanks for the huge offer to take over the maintance of geckodriver btw! :)

**Comment by szaimen** [comment]
It seem there are some blockers for using chromium: https://github.com/nextcloud/spreed/pull/9177#issuecomment-1667664311

**Comment by jospoortvliet** [comment]
> It seem there are some blockers for using chromium: [nextcloud/spreed#9177 (comment)](https://github.com/nextcloud/spreed/pull/9177#issuecomment-1667664311)

it's now merged ;-)

**Comment by nickvergessen** [comment]
While chromium recording is there, it is not the recommended solution

**Comment by szaimen** [comment]
> While chromium recording is there, it is not the recommended solution

I see. In that case, @Zoey2936, you offered a few comments above that you could take over the maintainership of the geckodriver package in alpine aports. Is that still the case?

**Comment by Zoey2936** [comment]
yes

**Comment by Zoey2936** [comment]
I've created https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/56841

**Comment by Zoey2936** [comment]
@szaimen can you please change the apk add line?

**Comment by szaimen** [comment]
> @szaimen can you please change the apk add line?

Will it be already available now or do we need to wait e.g. for Alpine 3.19.1?

**Comment by Zoey2936** [comment]
I think 1.20.0

**Comment by szaimen** [comment]
> I think 3.20.0

Okay, then I'd say lets wait for that

**Comment by szaimen** [comment]
> Yes but it moved from testing to community, so this needs to be changed in the apk add line

Okay, so we can already change that now or do we still need to wait for 3.19.1 for this to become available?

**Comment by Zoey2936** [comment]
alpine v3.20.1 was released

**Comment by szaimen** [comment]
Nice :)

**Comment by szaimen** [comment]
Done in https://github.com/nextcloud/all-in-one/pull/4857/files#diff-1f09bef912f390b61f65c4422014405c37d37c466cc37283db36a37464a1f907

**Comment by szaimen** [comment]
This is now released with v9.1.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
